common/src/main/java/io/druid/timeline/TimelineLookup.java

```diff
@@ -35,6 +35,8 @@
    */
   public Iterable<TimelineObjectHolder<VersionType, ObjectType>> lookup(Interval interval);
 
+  public Iterable<TimelineObjectHolder<VersionType, ObjectType>> lookup(Interval interval, boolean incompleteOk);
```

Comment:
`allowIncomplete` would be clearer I think. Maybe add some javadoc as well while we're at it.